xrpl-wasm-stdlib/src/core/types/vector_256.rs

```diff
@@ -0,0 +1,283 @@
+//! Vector256 - Array of 256-bit hashes
+//!
+//! This module provides support for XRPL's Vector256 field type, which represents
+//! an array of 32-byte (256-bit) hashes. Common uses include CredentialIDs, NFTokenOffers,
+//! and other fields that contain multiple hash values.
+//!
+//! ## Serialization Format
+//!
+//! Vector256 fields are serialized as:
+//! - 1 byte: total byte length of the data (must be a multiple of 32)
+//! - N × 32 bytes: the hash values
+//!
+//! The count of hashes is calculated as: data_length / 32
+//!
+//! ## Example
+//!
+//! ```no_run
+//! use xrpl_wasm_stdlib::core::types::vector_256::Vector256;
+//! use xrpl_wasm_stdlib::core::types::uint::Hash256;
+//!
+//! // Parse from raw bytes (64 = 2 hashes × 32 bytes each)
+//! let raw_data = [64, /* 64 bytes of hash data */];
+//! let vector = Vector256::from_bytes(&raw_data).unwrap();
+//! assert_eq!(vector.len(), 2);
+//! ```
+
+use crate::core::types::uint::Hash256;
+use crate::host::Error;
+
+/// Maximum number of hashes that can be stored in a Vector256
+pub const MAX_VECTOR_256_SIZE: usize = 8;
+
+/// A collection of 256-bit hashes with a maximum size
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[repr(C)]
+pub struct Vector256 {
+    /// The hash values
+    pub hashes: [Hash256; MAX_VECTOR_256_SIZE],
+    /// Number of valid hashes (0 to MAX_VECTOR256_SIZE)
+    pub len: u8,
+}
+
+impl Vector256 {
```

Comment:
I actually think this code is unnecessary and needs a rippled fix. This should be a type you should iterate through with a locator, instead of fetching the whole thing.